main/src/ca/uwaterloo/flix/language/ast/ops/TypedAstOps.scala

```diff
@@ -291,6 +300,23 @@ object TypedAstOps {
       case Pattern.Array(elms, tpe, loc) => elms.foldLeft(Map.empty[Symbol.VarSym, Type]) {
         case (macc, elm) => macc ++ binds(elm)
       }
+      case Pattern.ArrayTailSpread(elms, sym, tpe, loc) =>
+        val boundElms = elms.foldLeft(Map.empty[Symbol.VarSym, Type]) {
+          case (macc, elm) => macc ++ binds(elm)
+        }
+        sym match{
+          case None => boundElms
+          case Some(value) => Map(value -> tpe) ++ boundElms
+      }
+      case Pattern.ArrayHeadSpread(sym, elms, tpe, loc) =>
+        val boundElms = elms.foldLeft(Map.empty[Symbol.VarSym, Type]) {
+          case (macc, elm) => macc ++ binds(elm)
+        }
+        sym match{
+          case None => boundElms
+          case Some(value) => Map(value -> tpe) ++ boundElms
+      }
+
```

Comment:
Here is a slightly better way that we prefer to write these things:

```scala
val init = sym match {
  case None => Map.empty[Symbol.VarSym, Type]
  case Some(value) => Map(value -> tpe)
}
val boundElms = elms.foldLeft(init) {
  case (macc, elm) => macc ++ binds(elm)
}
```
Makes sense?